### ISSUE TITLE
bug: nav, getLength is publicly exported but not part of public types

### ISSUE THREAD

**Issue by aeharding** [ionic-team/ionic-framework#28826]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

v7.x

### Current Behavior

`ion-nav` does not have method `getLength()` typed.

### Expected Behavior

`ionNav.getLength()` is typed

### Steps to Reproduce

1. View stackblitz page-one.tsx.
2. Observe typescript error.
3. Open console.
4. Observe `1` printed.

### Code Reproduction URL

https://stackblitz.com/edit/aixnb2-innfhm?file=src%2Fpage-one.tsx

### Ionic Info

```
Ionic:

   Ionic CLI       : 7.1.1 (/Users/aeharding/.nvm/versions/node/v20.8.1/lib/node_modules/@ionic/cli)
   Ionic Framework : @ionic/react 7.6.3

Capacitor:

   Capacitor CLI      : 5.6.0
   @capacitor/android : 5.6.0
   @capacitor/core    : 5.6.0
   @capacitor/ios     : 5.6.0

Utility:

   cordova-res : not installed globally
   native-run  : not installed globally

System:

   NodeJS : v20.8.1 (/Users/aeharding/.nvm/versions/node/v20.8.1/bin/node)
   npm    : 10.1.0
   OS     : macOS Unknown
```

**Comment by liamdebeasi** [comment]
Hey there,

Can you clarify the bug report? `getLength` is not a public API, so it should not be part of the public type interfaces.

**Comment by aeharding** [comment]
Sure! So I use this in in order to call public APIs, such as `removeIndex`. I use this to remove the previous navigation page from the stack. Without knowing how long the current stack is I am not sure how to call `removeIndex`.

A lot of the public APIs for `ion-nav` rely on `index`.

If there is a public API for determining the length of the stack in order to call `removeIndex` please let me know :)

A practical example is after a user has signed up, I want to remove previous intermediate signup pages from the stack (since the user should not be able to navigate back). Example code:

```ts

    nav.push(
      () => <Joined verifyEmailSent={verify_email_sent} />,
      null,
      null,
      async (hasCompleted, requiresTransition, entering) => {
        // Remove signup steps from stack (everything between root and current nav view)
        // (user should not be able to navigate back after signup)

        // TODO open bug for missing ionic type
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        if (entering) nav.removeIndex(1, (nav as any).getLength() - 2);
      },
    );
```

**Comment by liamdebeasi** [comment]
Ah I see, that makes sense. It looks like the `@Method` decorator was accidentally removed from `getLength`: https://github.com/ionic-team/ionic-framework/commit/1d46973785aa204e13a44a740c9df75751d9f88d#diff-4b2ac275268173207c99a590b0ced8b67dc3d697815eb51a3c0546ee7cb0c0aeR429-L445

I'll put up a PR to fix this.

**Comment by liamdebeasi** [comment]
Thanks for the issue. This has been resolved via https://github.com/ionic-team/ionic-framework/pull/28832, and a fix will be available in an upcoming release of Ionic Framework. By marking `getLength` as a public method I also needed to make it return `Promise<number>` instead of `number`. Since this API was technically part of the public API (even though it's hard to use as-is right now), we are considering this a breaking change to minimize disruptions to existing apps. This fix will be available in the next major release of Ionic.
